documentation-website/Writerside/topics/Working-with-Tables.topic

```diff
@@ -2,44 +2,59 @@
-            <p>
-                Exposed supports a variety of <a href="Data-Types.topic">column types</a>, including <code>integer</code>,
-                <code>varchar</code>,<code>bool</code>, and more. Each column is defined by calling the appropriate
-                method on the <code>Table</code> object.
-            </p>
-            <p>The following example defines a simple table with an integer <code>id</code> column and a string
-                <code>name</code> column:</p>
-            <code-block lang="kotlin">
-                object Cities : Table() {
-                    val id = integer("id")
-                    val name = varchar("name", 50)
-                }
-            </code-block>
+        <chapter title="Table" id="table-class">
+            <include from="DSL-Table-Types.topic" element-id="dsl-table-type"/>
+        </chapter>
+        <chapter title="IdTable" id="id-table-type">
+            <include from="DAO-Table-Types.topic" element-id="dao-table-types"/>
+            For more information on <code>IdTable</code> types, see <a href="DAO-Table-Types.topic">DAO
+            Table Types</a>.
+        </chapter>
+    </chapter>
+    <chapter title="Defining tables" id="defining-tables">
+        <p>
+            A database table is represented by an object inherited from a table class.
+        </p>
+        <code-block lang="kotlin">
+            object Cities : Table() {}
+        </code-block>
+        <p>
+            Exposed supports a variety of <a href="Data-Types.topic">column types</a>, including <code>integer</code>,
+            <code>varchar</code>,<code>bool</code>, and more. Each column is defined by calling the appropriate
+            method on the <code>Table</code> object.
+        </p>
+        <p>The following example defines a table with an auto-incrementing integer <code>id</code> column and a string
+            <code>name</code> column:</p>
+        <tabs group="type">
+            <tab title="Table" id="dsl-table-def" group-key="Table">
+                <include from="DSL-Table-Types.topic" element-id="dsl-table-example"/>
+            </tab>
+            <tab title="IntIdTable" id="dao-table-def" group-key="IntIdTable">
+                <include from="DAO-Table-Types.topic" element-id="dao-table-example"/>
+            </tab>
+        </tabs>
+        <p>
+            Creating the table with the above definition will result in the following SQL equivalent:
+        </p>
+        <tabs group="type">
+            <tab title="Table" id="dsl-create-table-sql" group-key="Table">
+                <include from="DSL-Table-Types.topic" element-id="dsl-table-example-sql"/>
+            </tab>
+            <tab title="IntIdTable" id="dao-create-table-sql" group-key="IntIdTable">
+                <include from="DAO-Table-Types.topic" element-id="dao-table-example-sql"/>
+            </tab>
+        </tabs>
+        <chapter title="Configuring a custom table name" id="name-from-object">
```

Comment:
Just a thought about content for the future (outside the scope of this PR):

We should probably have a section similar to this for all Exposed naming conventions during table definition, but at minimum definitely for column names.
For example, quite a few sample code blocks use a column registered as `varchar("name", 50)`, so when we show related SQL (using H2) it translates to `"name" VARCHAR(50)`. Users may not get the implication that Exposed auto-quotes (& retains case-sensitivity of) all keywords used as names. This would probably be a prudent addition somewhere later.

Reply:
Good idea, I created a new issue for this at [EXPOSED-571](https://youtrack.jetbrains.com/issue/EXPOSED-571) so we don't forget.